### PR TITLE
Add `appinsights-props` attribute

### LINKED ISSUE #4: Make `data-appinsights` attribute an Aurelia attribute
To be more aurelia-like, what do you think of converting the `data-appinsights` attributes to good old fashion Aurelia custom attributes? Should be simple :)

### PR DESCRIPTION
**Pull request by RichiCoder1**
Allows for associating arbitrary objects with click tracking.  Closes #4.

### CODE REVIEW

**Comment by RichiCoder1** [comment]
A lot happening here, so will happily take questions or comments.

File: src/appinsights-props.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export var dataKey = 'appinsights';
+
+export class AppinsightsPropsCustomAttribute {
+	static inject = [Element];
```
Comment (miguelzakharia) [inline src/appinsights-props.js:4]: Please use `@inject()` instead so it stays consistent with the other code.

File: src/appinsights-props.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+	constructor(element) {
+		this.element = element;
+	}
+
+	valueChanged(newValue) {
+		if (!(typeof newValue == 'object' && newValue !== null)) {
```
Comment (miguelzakharia) [inline src/appinsights-props.js:11]: Consider `if(typeof newValue !== 'object' || newValue === null)` instead to make it easier to read.

**Review by miguelzakharia — CHANGES_REQUESTED** [review]
Two really minor things otherwise it's good to go.

File: src/appinsights-props.js (hunk 1 above)
Comment (RichiCoder1) [reply src/appinsights-props.js:4]: Oops. Yup. For some reason I wasn't sure if I could use decorators.

**Review by miguelzakharia — APPROVED** [review]

**Comment by miguelzakharia** [comment]
I'll publish this to NPM later today or tomorrow.

**Pull request merged**